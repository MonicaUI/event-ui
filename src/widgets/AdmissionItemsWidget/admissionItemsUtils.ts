import { AdmissionItem } from '@cvent/event-ui-apollo-server/src/schema/types';

/**
 * builds a filter for admissionItems based on selected registration typeIds to query graphQL
 * if invitee has selected regTypeId1 and regTypeId2 for his guest, then the filter will look like
 * applicableContactTypes is empty or applicableContactTypes contains 'regTypeA' or applicableContactTypes contains 'regTypeB'
 */
export const buildAdmissionItemsFilter = (registrationTypeIds: Array<string>): string => {
  return (
    'applicableContactTypes is empty' +
    (registrationTypeIds.length > 0
      ? ' or ' + registrationTypeIds.map(id => `applicableContactTypes contains '${id}'`).join(' or ')
      : '')
  );
};

/**
 * get all admissionItems which are visible to primary invitee
 */
export const getPrimaryVisibleAdmissionItems = (
  admissionItems: Array<AdmissionItem>,
  primaryRegistrationTypeId: string
): Array<AdmissionItem> => {
  return (admissionItems || []).filter(admissionItem =>
    admissionItem.applicableContactTypes.includes(primaryRegistrationTypeId)
  );
};
