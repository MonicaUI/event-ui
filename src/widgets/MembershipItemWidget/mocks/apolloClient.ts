import { DESELECT_MEMBERSHIP, SELECT_MEMBERSHIP } from '../updateMembershipQuery';

const selectMembershipRequest = {
  query: SELECT_MEMBERSHIP,
  variables: {
    regCartId: '53372990-b26e-4556-8c42-efee05473e29',
    membershipItemId: '457b1725-6394-4476-92b2-6ecfb106791f',
    productId: '95bf4765-4706-425f-bb5a-bd3ba83075b9',
    eventRegistrationId: '00000000-0000-0000-0000-000000000001',
    renewal: false
  }
};

const deselectMembershipRequest = {
  query: DESELECT_MEMBERSHIP,
  variables: {
    regCartId: '53372990-b26e-4556-8c42-efee05473e29',
    membershipItemId: '457b1725-6394-4476-92b2-6ecfb106791f',
    productId: '95bf4765-4706-425f-bb5a-bd3ba83075b9',
    eventRegistrationId: '00000000-0000-0000-0000-000000000001',
    renewal: false
  }
};

const regCartResponse = {
  eventId: 'd780d258-4b8d-422f-a686-fd43862f2d09',
  localeId: 1033,
  accountSnapshotVersion: 'XQlU4Bg_NkiqdRIYerVqOyBN_2RIrlCM',
  eventSnapshotVersions: {
    'f694fb1c-a278-4555-ae2b-73042fda2063': 'Opbs0Iz4WlyFxIVggb1vIg7DXC.dfMFj'
  },
  regCartId: '53372990-b26e-4556-8c42-efee05473e29',
  lastSavedPageId: 'regProcessStep1',
  status: 'INPROGRESS',
  groupRegistration: false,
  volumeDiscountsInCartStatusType: 'NO_VOLUME_DISCOUNT_IN_CART',
  eventRegistrations: [
    {
      eventRegistrationId: '00000000-0000-0000-0000-000000000001',
      attendee: {
        personalInformation: {
          firstName: 'Alexandra',
          lastName: 'Petersen',
          emailAddress: 'rotemobeb@mailinator.com',
          company: 'Davidson and Copeland Inc',
          title: 'Accusamus et sunt incidunt officia ut repellendus Sit ut',
          mobilePhone: 'Id voluptatem laborum et quos',
          customFields: [],
          emailAddressDomain: 'mailinator.com'
        },
        isGroupMember: false,
        eventAnswers: []
      },
      attendeeType: 'ATTENDEE',
      displaySequence: 1,
      productRegistrations: [],
      requestedAction: 'REGISTER',
      externalRegistrationContactId: '',
      eventId: 'd780d258-4b8d-422f-a686-fd43862f2d09',
      primaryRegistrationId: '00000000-0000-0000-0000-000000000000',
      sessionRegistrations: {},
      quantityItemRegistrations: {},
      donationItemRegistration: {},
      membershipItemRegistrations: {},
      sessionWaitlists: {},
      sessionBundleRegistrations: {},
      registrationTypeId: '78ba0e75-7dab-4e38-a074-9e249ba8cd54',
      registrationPathId: '73386686-09c5-4e52-be42-43311c3bfc90',
      addGuestFromRelatedContacts: false,
      autoAssignRegTypeForEventRegistration: false,
      attendingFormatId: 0
    }
  ],
  isAdmin: false,
  registrationApprovalRequired: false,
  hasTravel: false,
  partial: true,
  regDecline: false,
  regWaitList: false,
  regApproval: false,
  regCancel: false,
  regMod: false,
  postRegPayment: false
};

export const mockMembershipSelectionSuccess = {
  request: selectMembershipRequest,
  newData: jest.fn(() => ({
    data: {
      selectMembershipItem: {
        regCart: {
          validationMessages: [],
          regCart: {
            ...regCartResponse,
            eventRegistrations: [
              {
                ...regCartResponse.eventRegistrations[0],
                membershipItemRegistrations: {
                  '95bf4765-4706-425f-bb5a-bd3ba83075b9': {
                    productType: 'MembershipItem',
                    requestedAction: 'REGISTER',
                    membershipItemId: '457b1725-6394-4476-92b2-6ecfb106791f',
                    renewal: false,
                    registrationTypeIdBeforeMembershipSelection: '00000000-0000-0000-0000-000000000000',
                    productId: '95bf4765-4706-425f-bb5a-bd3ba83075b9'
                  }
                }
              }
            ]
          }
        },
        membershipItemRegistrations: {
          '95bf4765-4706-425f-bb5a-bd3ba83075b9': {
            productType: 'MembershipItem',
            requestedAction: 'REGISTER',
            membershipItemId: '457b1725-6394-4476-92b2-6ecfb106791f',
            renewal: false,
            registrationTypeIdBeforeMembershipSelection: '00000000-0000-0000-0000-000000000000',
            productId: '95bf4765-4706-425f-bb5a-bd3ba83075b9'
          }
        }
      }
    }
  }))
};
export const mockMembershipDeselectionSuccess = {
  request: deselectMembershipRequest,
  newData: jest.fn(() => ({
    data: {
      deselectMembershipItem: {
        regCart: {
          validationMessages: [],
          regCart: {
            ...regCartResponse
          }
        }
      }
    }
  }))
};
