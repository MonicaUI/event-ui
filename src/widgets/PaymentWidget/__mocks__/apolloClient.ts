import { MOCK_GET_REG_CART_PRICING } from './useCachedRegCartPricing';

export const mockApolloClient = (readFragmentData = {}): $TSFixMe => {
  return {
    cache: {
      evict: jest.fn(),
      gc: jest.fn()
    },
    readFragment: jest.fn(() => readFragmentData),
    query: jest.fn()
  };
};

export const getApolloClientMocks = (state = {}): $TSFixMe => {
  // @ts-expect-error ts-migrate(2339) FIXME: Property 'registrationForm' does not exist on type... Remove this comment to see the full error message
  const { registrationForm: { regCart: { regCartId = '' } = {} } = {}, regCartPricing } = state;
  return [
    {
      request: {
        query: MOCK_GET_REG_CART_PRICING,
        /*
         * The variables must match exactly with those passed in by the components being tested
         * in order for the mock Apollo Client to return the result below.
         */
        variables: {
          regCartId,
          paymentInfo: JSON.stringify({ overriddenProductFees: {}, overriddenProductRefunds: {} }),
          lastSavedRegCart: JSON.stringify({ paymentInfo: null, lastSavedPageId: null }),
          travelCart: JSON.stringify({})
        }
      },
      result: {
        data: {
          pricing: {
            regCartPricing
          }
        }
      }
    }
  ];
};
