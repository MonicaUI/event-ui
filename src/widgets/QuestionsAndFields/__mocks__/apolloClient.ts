import { MockedResponse } from '@apollo/client/testing';
import { gqlQuery } from './pageVarietyPathQueryHooks';

type QuestionWidgetMockData = {
  isAirActualWidgetPresent: boolean;
  isAirRequestWidgetPresent: boolean;
  isGroupFlightWidgetPresent: boolean;
  isHotelRequestWidgetPresent: boolean;
};

export const getApolloClientMocks = (state: QuestionWidgetMockData): MockedResponse<Record<string, $TSFixMe>>[] => {
  const {
    isAirActualWidgetPresent = false,
    isAirRequestWidgetPresent = false,
    isGroupFlightWidgetPresent = false,
    isHotelRequestWidgetPresent = false
  } = state;
  return [
    {
      request: {
        query: gqlQuery
      },
      result: {
        data: {
          event: {
            registrationPath: {
              registration: {
                airActual: {
                  enabled: isAirActualWidgetPresent
                },
                airRequest: {
                  enabled: isAirRequestWidgetPresent
                },
                groupFlight: {
                  enabled: isGroupFlightWidgetPresent
                },
                hotelRequest: {
                  enabled: isHotelRequestWidgetPresent
                }
              }
            }
          }
        }
      }
    }
  ];
};
