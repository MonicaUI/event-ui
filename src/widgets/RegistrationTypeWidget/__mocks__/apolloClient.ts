import { GET_REG_TYPES } from './useRegTypes';

export const mockApolloClient = (readFragmentData = {}): $TSFixMe => {
  return {
    cache: {
      evict: jest.fn(),
      gc: jest.fn()
    },
    readFragment: jest.fn(() => readFragmentData)
  };
};

const request = {
  query: GET_REG_TYPES,
  variables: {
    eventId: 'EVENT_ID',
    regPathId: 'REG_PATH_ID',
    regCartId: 'REG_CART_ID',
    eventSnapshotVersion: 'VERSION',
    environment: 'ENV',
    attendeeType: 'INVITEE'
  }
};

export const getRegistrationTypeResults = [
  {
    request,
    result: {
      data: {
        getEvent: {
          visible: 'AVAILABLE',
          registrationTypes: [
            {
              id: '7a14802a-8aa0-4463-95f1-5e5793a63a8a',
              name: 'r1',
              visible: 'CAPACITY_FULL'
            },
            {
              id: '158ab11c-bc65-450a-92f6-6f5dc3a052ba',
              name: 'r2',
              visible: 'CLOSED'
            },
            {
              id: '00000000-0000-0000-0000-000000000000',
              name: '',
              visible: 'AVAILABLE'
            }
          ]
        }
      }
    }
  }
];

export const getEventData = [...getRegistrationTypeResults];
