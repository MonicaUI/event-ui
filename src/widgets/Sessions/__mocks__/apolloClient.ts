import { REGISTER_SESSION_BUNDLE } from '../../../redux/registrationForm/regCart/sessionBundles';

const request = {
  query: REGISTER_SESSION_BUNDLE,
  variables: {
    regCartId: '01d4655d-f010-458d-8a10-a408f97c029c',
    input: [
      {
        eventRegistrationId: '00000000-0000-0000-0000-000000000001',
        productId: '3a53a9c0-f8f9-451a-919a-0ab577e9fb64',
        requestedAction: 'REGISTER',
        registrationSourceType: 'Selected'
      }
    ]
  }
};

export const mockRegisterSessionBundleSuccess = {
  request,
  newData: jest.fn(() => ({
    data: {
      response: {
        regCart: {
          sendEmail: true,
          eventId: 'f694fb1c-a278-4555-ae2b-73042fda2063',
          localeId: 1033,
          accountSnapshotVersion: 'XQlU4Bg_NkiqdRIYerVqOyBN_2RIrlCM',
          eventSnapshotVersions: {
            'f694fb1c-a278-4555-ae2b-73042fda2063': 'Opbs0Iz4WlyFxIVggb1vIg7DXC.dfMFj'
          },
          regCartId: '01d4655d-f010-458d-8a10-a408f97c029c',
          lastSavedPageId: 'regPage:617b1dc3-3ddb-4468-8087-049a4e6e51e8',
          status: 'INPROGRESS',
          dequeueStatus: {
            totalSteps: 0,
            currentStep: 0
          },
          groupRegistration: false,
          volumeDiscountsInCartStatusType: 'NO_VOLUME_DISCOUNT_IN_CART',
          eventRegistrations: [
            {
              eventRegistrationId: '00000000-0000-0000-0000-000000000001',
              attendee: {
                personalInformation: {
                  firstName: 'invitee',
                  lastName: '1',
                  emailAddress: 'invitee@j.mail',
                  customFields: [],
                  emailAddressDomain: 'j.mail'
                },
                isGroupMember: false,
                eventAnswers: []
              },
              attendeeType: 'ATTENDEE',
              displaySequence: 1,
              productRegistrations: [
                {
                  requestedAction: 'REGISTER',
                  productType: 'AdmissionItem',
                  productId: '6a0ee119-0fd3-4e70-a61e-16361ebef51f',
                  quantity: 1
                }
              ],
              requestedAction: 'REGISTER',
              externalRegistrationContactId: '',
              contactNoMatchInSfCampaign: false,
              eventId: 'f694fb1c-a278-4555-ae2b-73042fda2063',
              primaryRegistrationId: '00000000-0000-0000-0000-000000000000',
              sessionRegistrations: {
                '0c7af723-a761-447e-af49-4c6654de14ed': {
                  requestedAction: 'REGISTER',
                  productId: '0c7af723-a761-447e-af49-4c6654de14ed',
                  registrationSourceType: 'Track',
                  registrationSourceParentId: '3a53a9c0-f8f9-451a-919a-0ab577e9fb64',
                  includedInAgenda: false
                }
              },
              quantityItemRegistrations: {},
              donationItemRegistrations: {},
              sessionWaitlists: {},
              sessionBundleRegistrations: {
                '3a53a9c0-f8f9-451a-919a-0ab577e9fb64': {
                  requestedAction: 'REGISTER',
                  productId: '3a53a9c0-f8f9-451a-919a-0ab577e9fb64'
                }
              },
              registrationTypeId: '00000000-0000-0000-0000-000000000000',
              registrationPathId: 'c6da5ffa-fa0d-465d-823a-8f4c328f7619',
              addGuestFromRelatedContacts: false,
              autoAssignRegTypeForEventRegistration: false,
              attendingFormatId: 0
            },
            {
              eventRegistrationId: '9a44e12c-bc20-43fc-88ad-36c42c8de107',
              attendee: {
                personalInformation: {
                  firstName: 'guest',
                  lastName: '1',
                  emailAddress: 'guest@j.mail',
                  customFields: [],
                  emailAddressDomain: 'j.mail'
                },
                isGroupMember: false,
                eventAnswers: []
              },
              attendeeType: 'GUEST',
              displaySequence: 1,
              productRegistrations: [
                {
                  requestedAction: 'REGISTER',
                  productType: 'AdmissionItem',
                  productId: '6a0ee119-0fd3-4e70-a61e-16361ebef51f',
                  quantity: 1
                }
              ],
              requestedAction: 'REGISTER',
              contactNoMatchInSfCampaign: false,
              eventId: 'f694fb1c-a278-4555-ae2b-73042fda2063',
              primaryRegistrationId: '00000000-0000-0000-0000-000000000001',
              sessionRegistrations: {
                '0c7af723-a761-447e-af49-4c6654de14ed': {
                  requestedAction: 'REGISTER',
                  productId: '0c7af723-a761-447e-af49-4c6654de14ed',
                  registrationSourceType: 'Track',
                  registrationSourceParentId: '3a53a9c0-f8f9-451a-919a-0ab577e9fb64',
                  includedInAgenda: false
                }
              },
              quantityItemRegistrations: {},
              donationItemRegistrations: {},
              sessionWaitlists: {},
              sessionBundleRegistrations: {
                '3a53a9c0-f8f9-451a-919a-0ab577e9fb64': {
                  requestedAction: 'REGISTER',
                  productId: '3a53a9c0-f8f9-451a-919a-0ab577e9fb64'
                }
              },
              registrationTypeId: '00000000-0000-0000-0000-000000000000',
              registrationPathId: 'c6da5ffa-fa0d-465d-823a-8f4c328f7619',
              addGuestFromRelatedContacts: false,
              autoAssignRegTypeForEventRegistration: false,
              attendingFormatId: 0
            }
          ],
          isAdmin: false,
          registrationApprovalRequired: false,
          hasTravel: false,
          partial: true,
          regApproval: false,
          regCancel: false,
          regMod: false,
          regDecline: false,
          regWaitList: false,
          postRegPayment: false
        },
        validationMessages: []
      }
    }
  }))
};

export const mockRegisterSessionBundleError = {
  request,
  error: new Error('Some error occurred')
};

export const mockRegisterSessionBundleCapacityReachError = {
  request,
  error: {
    statusCode: 422,
    result: {
      validationMessages: [
        {
          severity: 'Error',
          localizationKey: 'REGAPI.CAPACITY_UNAVAILABLE',
          parametersMap: {
            eventRegistrationId: 'evtRegId',
            productId: 'sessionBundleId',
            productType: 'Track',
            registrationTypeId: 'attendee'
          }
        }
      ]
    }
  }
};
