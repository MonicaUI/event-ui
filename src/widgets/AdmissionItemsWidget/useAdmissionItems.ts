import { gql, useQuery } from '@apollo/client';
import { useSelector, useStore } from 'react-redux';
import { values } from 'lodash';
import { getRegistrationPathIdForWidget } from 'event-widgets/redux/selectors/website/pluginData/registrationProcessNavigation';
import { buildAdmissionItemsFilter } from './admissionItemsUtils';
import {
  getHasCurrentRegistrationAtLeastOneInPersonAttendee,
  getHasCurrentRegistrationAtLeastOneVirtualAttendee
} from '../../redux/selectors/hybridEventSelectors';

export const GET_ADMISSION_ITEMS = gql`
  query getEvents(
    $eventId: ID!
    $eventSnapshotVersion: String!
    $regPathId: ID!
    $environment: String!
    $regCartId: String
    $filter: Filter
    $includeInPersonCapacity: Boolean!
    $includeVirtualCapacity: Boolean!
  ) {
    eventId @client @export(as: "eventId")
    environment @client @export(as: "environment")
    eventSnapshotVersion @client @export(as: "eventSnapshotVersion")
    getEvent(
      input: {
        eventId: $eventId
        eventSnapshotVersion: $eventSnapshotVersion
        regPathId: $regPathId
        environment: $environment
        regCartId: $regCartId
        filter: $filter
      }
    ) {
      admissionItems {
        id
        name
        visible
        description
        code
        applicableContactTypes
        defaultFeeId
        displayOrder
        capacity {
          inPerson @include(if: $includeInPersonCapacity) {
            available
            total
          }
          virtual @include(if: $includeVirtualCapacity) {
            available
            total
          }
        }
      }
    }
  }
`;

export function useAdmissionItems(widgetId: $TSFixMe, registrationTypeIds: $TSFixMe): $TSFixMe {
  const store = useStore();
  const state = store.getState();
  const regCartId = useSelector(s => (s as $TSFixMe).userSession.regCartId);
  const regPathId = getRegistrationPathIdForWidget(state, widgetId);

  const filter = buildAdmissionItemsFilter(registrationTypeIds || []);

  const includeInPersonCapacity = useSelector(getHasCurrentRegistrationAtLeastOneInPersonAttendee);
  const includeVirtualCapacity = useSelector(getHasCurrentRegistrationAtLeastOneVirtualAttendee);

  const query = useQuery(GET_ADMISSION_ITEMS, {
    variables: {
      regPathId,
      regCartId,
      filter,
      includeInPersonCapacity,
      includeVirtualCapacity
    }
  });

  return query.data?.getEvent?.admissionItems && values(query.data.getEvent.admissionItems);
}
