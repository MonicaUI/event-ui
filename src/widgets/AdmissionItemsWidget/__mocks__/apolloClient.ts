import { GET_ADMISSION_ITEMS } from '../useAdmissionItems';
import { Visible } from '@cvent/event-ui-apollo-server/src/schema/types';

export const eventId = 'EVENT_ID';
export const environment = 'S437';
export const eventSnapshotVersion = 'VERSION';

export const mockApolloClient = [
  {
    request: {
      query: GET_ADMISSION_ITEMS,
      variables: {
        eventId,
        environment,
        eventSnapshotVersion,
        regCartId: 'REG_CART_ID',
        regPathId: 'REG_PATH_ID',
        filter:
          "applicableContactTypes is empty or applicableContactTypes contains '00000000-0000-0000-0000-000000000000'",
        includeInPersonCapacity: true,
        includeVirtualCapacity: false
      }
    },
    result: {
      data: {
        getEvent: {
          admissionItems: [
            {
              id: '213994f0-f2e1-48da-a016-37a80a5cc5d6',
              name: 'Event Registration',
              visible: Visible.Available,
              description: '',
              code: '',
              applicableContactTypes: ['regTypeId1'],
              defaultFeeId: '00000000-0000-0000-0000-000000000000',
              displayOrder: 2,
              capacity: {
                inPerson: {
                  available: -1,
                  total: -1
                }
              }
            }
          ]
        }
      }
    }
  }
];
