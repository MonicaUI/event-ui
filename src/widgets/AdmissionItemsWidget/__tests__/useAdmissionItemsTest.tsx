import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MockedProvider } from '@apollo/client/testing';
import { getRegistrationPathIdForWidget } from 'event-widgets/redux/selectors/website/pluginData/registrationProcessNavigation';
import { buildAdmissionItemsFilter } from '../admissionItemsUtils';
import { act } from 'react-dom/test-utils';
import { useAdmissionItems } from '../useAdmissionItems';
import configureStore from '../../../redux/configureStore';
// eslint-disable-next-line jest/no-mocks-import
import { mockApolloClient, eventId, eventSnapshotVersion, environment } from '../__mocks__/apolloClient';
import { InMemoryCache } from '@apollo/client';
import { mapValues } from 'lodash';

jest.mock('event-widgets/redux/selectors/website/pluginData/registrationProcessNavigation', () => ({
  getRegistrationPathIdForWidget: jest.fn()
}));

jest.mock('../admissionItemsUtils', () => ({
  buildAdmissionItemsFilter: jest.fn()
}));

// @ts-expect-error ts-migrate(2339) FIXME: Property 'mockImplementation' does not exist on ty... Remove this comment to see the full error message
getRegistrationPathIdForWidget.mockImplementation(() => 'REG_PATH_ID');
(buildAdmissionItemsFilter as $TSFixMe).mockImplementation(
  () => "applicableContactTypes is empty or applicableContactTypes contains '00000000-0000-0000-0000-000000000000'"
);

const getState = () => {
  return {
    environment: 'S437',
    eventSnapshotVersion: 'VERSION',
    appData: {
      registrationSettings: {
        registrationPaths: {
          REG_PATH_ID: {
            guestRegistrationSettings: {
              isGuestRegistrationEnabled: true
            }
          }
        }
      }
    },
    registrationForm: {
      currentEventRegistrationId: '00000000-0000-0000-0000-000000000001',
      regCart: {
        volumeDiscountsInCartStatusType: 'NO_VOLUME_DISCOUNT_IN_CART',
        eventRegistrations: {
          '00000000-0000-0000-0000-000000000001': {
            eventRegistrationId: '00000000-0000-0000-0000-000000000001',
            primaryRegistrationId: '00000000-0000-0000-0000-000000000000',
            attendee: {
              personalInformation: {
                firstName: 'm',
                lastName: 'd'
              }
            },
            attendeeType: 'ATTENDEE',
            displaySequence: 1,
            eventId: '1efacd58-d0aa-4984-929e-7a65648270bb',
            registrationPathId: 'REG_PATH_ID',
            requestedAction: 'REGISTER',
            sessionRegistrations: {
              '3253430c-d19a-421b-9592-70b8b2b7c4c5': {
                includedInAgenda: false,
                productId: '3253430c-d19a-421b-9592-70b8b2b7c4c5',
                registrationSourceType: 'Included',
                requestedAction: 'REGISTER'
              },
              'bd0d311c-68a8-4706-bee1-2ca42023a339': {
                includedInAgenda: false,
                productId: 'bd0d311c-68a8-4706-bee1-2ca42023a339',
                registrationSourceType: 'Selected',
                requestedAction: 'REGISTER'
              }
            },
            attendingFormatId: 0
          },
          '55e6409a-1f00-4c67-b586-9666229d9dc9': {
            eventRegistrationId: '55e6409a-1f00-4c67-b586-9666229d9dc9',
            primaryRegistrationId: '00000000-0000-0000-0000-000000000001',
            attendee: {
              personalInformation: {
                firstName: 'abc',
                lastName: 'Guest 02'
              }
            },
            attendeeType: 'GUEST',
            displaySequence: 2,
            eventId: '1efacd58-d0aa-4984-929e-7a65648270bb',
            registrationPathId: 'REG_PATH_ID',
            requestedAction: 'REGISTER',
            sessionRegistrations: {
              '3253430c-d19a-421b-9592-70b8b2b7c4c5': {
                includedInAgenda: false,
                productId: '3253430c-d19a-421b-9592-70b8b2b7c4c5',
                registrationSourceType: 'Included',
                requestedAction: 'REGISTER'
              },
              'bd0d311c-68a8-4706-bee1-2ca42023a339': {
                includedInAgenda: false,
                productId: 'bd0d311c-68a8-4706-bee1-2ca42023a339',
                registrationSourceType: 'Selected',
                requestedAction: 'REGISTER'
              }
            },
            attendingFormatId: 0
          }
        },
        groupRegistration: false,
        regCartId: 'REG_CART_ID',
        status: 'INPROGRESS'
      }
    },
    event: {
      id: 'EVENT_ID',
      attendingFormat: 0
    },
    userSession: {
      regCartId: 'REG_CART_ID'
    }
  };
};

async function dispatch(action) {
  if (typeof action === 'function') {
    await action(dispatch, getState);
  }
}

const subscribe = () => {};
const defaultProps = {
  classes: {},
  style: {},
  translate: (res, opts) => (opts ? `${res}:${JSON.stringify(opts)}` : res),
  config: {
    headerText: 'Header',
    instructionalText: 'Instructional text',
    display: {
      capacity: true,
      description: true,
      fees: true,
      itemCode: true
    }
  },
  isRegistrationPage: true,
  id: 'widgetId',
  'data-cvent-id': 'widgetId',
  store: { dispatch, getState, subscribe },
  layout: {
    cellSize: 4
  }
};

const waitWithAct = async () => {
  // Act is used because the Redux store is updated during wait
  return act(async () => {
    await new Promise(resolve => setTimeout(resolve, 100));
  });
};

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const TestComponent = _props => {
  const admissionItems = useAdmissionItems('widgetId', []);
  return <pre>{JSON.stringify(admissionItems, null, 2)}</pre>;
};

function createCache(store, constants) {
  return new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          ...mapValues(constants, value => ({
            read() {
              return value;
            }
          }))
        }
      }
    },
    addTypename: false
  });
}

const mountComponent = async () => {
  const store = configureStore(getState(), {});
  const component = mount(
    <Provider store={store}>
      <MockedProvider
        mocks={mockApolloClient}
        cache={createCache(store, {
          eventId,
          environment,
          eventSnapshotVersion
        })}
        addTypename={false}
      >
        <TestComponent {...defaultProps} />
      </MockedProvider>
    </Provider>
  );
  // Wait for Apollo Client MockedProvider to render mock query results
  await waitWithAct();
  await component.update();
  return component;
};

describe('useAdmissionItems', () => {
  it('should render with query result', async () => {
    const component = await mountComponent();
    expect(component).toMatchSnapshot();
  });
});
