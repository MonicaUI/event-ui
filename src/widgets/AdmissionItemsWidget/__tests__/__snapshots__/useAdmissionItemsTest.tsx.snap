// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useAdmissionItems should render with query result 1`] = `
<pre>
  [
  {
    "applicableContactTypes": [
      "regTypeId1"
    ],
    "capacity": {
      "inPerson": {
        "available": -1,
        "total": -1
      }
    },
    "code": "",
    "defaultFeeId": "00000000-0000-0000-0000-000000000000",
    "description": "",
    "displayOrder": 2,
    "id": "213994f0-f2e1-48da-a016-37a80a5cc5d6",
    "name": "Event Registration",
    "visible": "AVAILABLE"
  }
]
</pre>
`;
